client/state/use-reducer.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { createElement, Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { addReducer, removeReducer } from 'state';
+
+/**
+ * Higher Order Component to add a reducer to the store's reducer list.
+ * TODO: Add this file to a place accessible by all extensions when it's ready.
+ *
+ * @param { string } reducerName The name for the reducer/key for redux state tree.
+ * @param { func } reducerFunc The function that takes (state, action) and returns state.
+ * @returns { func } A curried function that takes a component to wrap.
+ */
+export default function useReducer( reducerName, reducerFunc ) {
+	return function wrapWithUseReducer( WrappedComponent ) {
+		class UseReducer extends Component {
+			constructor( props, context ) {
+				super( props, context );
+				this.store = props.store || context.store;
+				this.reducerName = reducerName;
+				this.reducerFunc = reducerFunc;
+			}
+
+			componentDidMount() {
+				addReducer( reducerName, reducerFunc );
+			}
+
+			componentWillReceiveProps( newProps ) {
+				if ( newProps.store ) {
+					removeReducer( reducerName );
+					this.store = newProps.store;
```

Comment:
The `store` instance property is actively managed in this class (in `constructor` and here), yet it's seemingly never used, as `addReducer` and `removeReducer` are already bound to `state`'s `calypsoStore` anyway. At the very least, `store` will implicitly be carried in `context` or `props` into `createElement( WrappedComponent, this.props )`, but that doesn't require the setting of `this.store`. Or am I missing something?